tt_metal/api/kernel.hpp

```diff
@@ -0,0 +1,66 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "types.hpp"
+#include "tt_metal/impl/kernels/kernel_types.hpp"
+
+//==================================================
+//                 KERNEL EXECUTION
+//==================================================
+
+namespace tt::tt_metal{
+namespace v1 {
+
+/**
+ * @brief Sets runtime arguments for a kernel.
+ *
+ * @param program The program containing the kernel.
+ * @param kernel KernelHandle representing the kernel ID.
+ * @param core_spec Specifies the cores where the runtime arguments will be set.
+ * @param runtime_args The runtime arguments to be set.
+ */
+void SetRuntimeArgs(
+    const Program program,
+    KernelHandle kernel,
```

Comment:
`CreateKernel` api accepts Program as an argument, so I assume it ties the kernel to the program.

Do we have to pass Program handle here?
What happens if KernelHandle and ProgramHandle mismatch?
Can we prevent this class of errors?

Reply:
The v1::KernelHandle can be a strong type.  We can make the KernelHandle sufficient to get to the right Program.  Let's remove ProgramHandle from SetRuntimeArgs.

Reply:
Having inspected the v0 implementation, the program is only used to lookup the kernel. In v1 the KernelHandle alone should be sufficient for looking up the kernel, so we can remove the ProgramHandle as an argument from this API functions.